catalog/plugin/catalog-plugin-content-uri/src/main/java/org/codice/ddf/catalog/content/plugin/uri/ContentUriAccessPlugin.java

```diff
@@ -41,19 +45,31 @@ public UpdateRequest processPreUpdate(UpdateRequest input,
         for (Map.Entry<Serializable, Metacard> entry : input.getUpdates()) {
             Metacard existingMetacard = existingMetacards.get(entry.getKey()
                     .toString());
+
+            if (oneOrOtherUriNull(existingMetacard, entry.getValue())) {
+                throw new StopProcessingException(CANNOT_OVERWRITE_MESSAGE);
+            }
+
             if (entry.getValue()
                     .getResourceURI() != null && existingMetacard.getResourceURI() != null
                     && !existingMetacard.getResourceURI()
                     .equals(entry.getValue()
                             .getResourceURI())
                     && ContentItem.CONTENT_SCHEME.equals(existingMetacard.getResourceURI()
                     .getScheme())) {
-                throw new StopProcessingException("Cannot overwrite resource URI in content store.");
+                throw new StopProcessingException(CANNOT_OVERWRITE_MESSAGE);
             }
         }
         return input;
     }
 
+    private boolean oneOrOtherUriNull(Metacard existingMetacard, Metacard updatedMetacard) {
```

Comment:
Went ahead and added this check at the beginning of the `for` loop before checking anything else. Also added an additional test where both existing and updated URIs are content schema and equal.